content/src/service/access/AccessCheckerImpl.ts

```diff
@@ -1,90 +1,399 @@
+                        const pointerParts: string[] = pointer.split(',')
+                        if (pointerParts.length === 2) {
+                            const x: number = parseInt(pointerParts[0], 10)
+                            const y: number = parseInt(pointerParts[1], 10)
+
+                            // Check that the address has access
+                            const hasAccess = await this.checkParcelAccess(x, y, timestamp, ethAddress)
+                            if (!hasAccess) {
+                                errors.push(`The provided Eth Address does not have access to the following parcel: (${x},${y})`)
+                            }
+                        } else {
+                            errors.push(`Scene pointers should only contain two integers separated by a comma, for example (10,10) or (120,-45). Invalid pointer: ${pointer}`)
+                        }
                     }
-                }
-            }))
+                }))
 
         return errors
     }
 
-    private async checkParcelAccess(x: number, y: number, ethAddress: EthAddress): Promise<boolean> {
-
-        const accessURL = `${this.dclApiBaseUrl}/parcels/${x}/${y}/${ethAddress}/authorizations`
+    private async checkParcelAccess(x: number, y: number, timestamp: Timestamp, ethAddress: EthAddress): Promise<boolean> {
         try {
-            const responseJson = await this.fetchHelper.fetchJson(accessURL)
-            return responseJson.data.isUpdateAuthorized
-        } catch(e) {
-            AccessCheckerImpl.LOGGER.warn(`Failed to check parcel access. Error was ${e.message}`)
+            return await this.isParcelUpdateAuthorized(x, y, timestamp, ethAddress)
+        } catch (error) {
+            AccessCheckerImpl.LOGGER.error(`Error checking parcel access (${x}, ${y}, ${timestamp}, ${ethAddress})`, error)
         }
         return false
     }
 
-    private async checkProfileAccess(pointers: Pointer[], ethAddress: EthAddress): Promise<string[]> {
-        const errors: string[] = []
+    /**
+     * Checks if the address had deployment access to that coordinate at the specified time.
+     */
+    private async isParcelUpdateAuthorized(
+        x: number,
+        y: number,
+        timestamp: Timestamp,
+        ethAddress: EthAddress
+    ): Promise<boolean> {
+        /* You get direct access if you were the:
+         *   - owner
+         *   - operator
+         *   - update operator
+         * at that time
+         */
+        const parcel = await this.getParcel(x, y, timestamp)
 
-        if (pointers.length!=1) {
-            errors.push(`Only one pointer is allowed when you create a Profile. Received: ${pointers}`)
+        if (parcel.estates?.length > 0 && parcel.estates[0].estateId) {
+            // The parcel belongs to an estate
+            return this.isEstateUpdateAuthorized(
```

Comment:
It this ok? If the estate validations fails, shouldn't we still check the parcel access?

Reply:
No, because if the parcel belongs to an estate, all the checking should be done based on the estate, not the parcel.

Reply:
We're changing the logic here, granting the permission if it's provided by the estate but also if it's provided by the parcel.